app/services/blog_search.rb

```diff
@@ -0,0 +1,10 @@
+class BlogSearch
+	include ActionView::Helpers::SanitizeHelper
+	include ActiveModel::Model
```

Comment:
What are these two `include`s being used for? I can't seem to work it out, sorry!

Reply:
ActionView::Helpers::SanitizeHelper is not in use, sorry for that

Using attr_accessor is the reason why I include ActiveModel::Model, I want BlogSearch class behave like a ActiveRecord

Reply:
I am not sure my explanation is make sense or not, normally I will add ActiveModel::Model to use attr_accessor (just like to have a setter and getter)

Reply:
Ah, gotcha. You might be interested to know that in fact `attr_accessor` is a Ruby method, and has nothing to do with `ActiveModel::Model`!

https://ruby-doc.org/core-2.0.0/Module.html#method-i-attr_accessor

As proof, try this in `irb`:

```ruby
class Thing
  attr_accessor :my_prop
end

thing = Thing.new
thing.my_prop = "Hello World"
thing.my_prop
```

Reply:
Yes, thanks, I will take a deep look at that, really appreciate about that.

Reply:
Ah, it just occurred to me that you need `include ActiveModel::Model` here because you're passing in the controller params to the `.new` method, like you would with an AR object. My bad!